nextjs-app/app/_components/Header.tsx

```diff
@@ -1,30 +1,88 @@
+'use client'
+
+import { useState, useEffect } from "react";
 import Link from "next/link";
+import LogoSvg from './LogoSvg';
 
 export default function Header() {
-  return (
-    <header className="fixed z-50 h-24 inset-0 bg-white/80 flex items-center backdrop-blur-lg">
-      <div className="container py-6 sm:px-6">
-        <div className="flex items-center justify-between gap-5">
+  const [headerOffset, setHeaderOffset] = useState(0);
+  const [isScrolled, setIsScrolled] = useState(false);
+  const [topPosition, setTopPosition] = useState(0);
+  const [transitionProgress, setTransitionProgress] = useState(0);
 
+  useEffect(() => {
+    const handleScroll = () => {
+      const viewportHeight = window.innerHeight;
+      const scrollPercent = window.scrollY / viewportHeight;
+      
+      if (scrollPercent < 0.001) {  // No scroll
+        setHeaderOffset(0);
+        setIsScrolled(false);
+        setTopPosition(0);
+        setTransitionProgress(0);
+      } else if (scrollPercent > 0.15) {  // Full transition
+        setHeaderOffset(0);
+        setIsScrolled(true);
+        setTopPosition(-20);  // Fixed final position
+        setTransitionProgress(1);
+      } else {
+        // Smooth transition from start
+        const progress = scrollPercent / 0.15;
+        setTransitionProgress(progress);
+        setIsScrolled(progress > 0.5);
+        setTopPosition(-20 * progress);  // Smoothly move to -20px
+      }
+    };
 
-          <nav className="">
-            <ul
-              role="list"
-              className="flex items-center gap-4 md:gap-6 leading-5 text-sm md:text-base tracking-tight font-normal"
-            >
-              <li>
-                <Link className="" href="/online">
-                  Online
-                </Link>
-              </li>
-              <li>
-                <Link href="/offline" className="">
-                  Offline
-                </Link>
-              </li>
-            </ul>
-          </nav>
+    window.addEventListener('scroll', handleScroll);
+    return () => window.removeEventListener('scroll', handleScroll);
+  }, []);
 
+  return (
+    <header 
+      className={`w-full fixed left-0 z-50 transition-all duration-300 ${
```

Comment:
try `sticky top-0` in place of fixed
https://developer.mozilla.org/en-US/docs/Web/CSS/position